src/overlays/actors/ovl_En_Time_Tag/z_en_time_tag.c

```diff
@@ -100,205 +113,230 @@ void func_80AC9FE4(EnTimeTag* this, PlayState* play) {
                             ActorCutscene_Stop(this->actor.cutscene);
                         }
                         break;
 
-                    case 0x122B:
+                    case 0x122B: // Mikau diary part 1
                         func_80152434(play, 0x3F);
-                        this->actionFunc = func_80ACA3C0;
-                        this->actor.home.rot.z = 0;
+                        this->actionFunc = EnTimeTag_Diary_TeachEvanSongSnippets;
+                        TIMETAG_DIARY_SONG(&this->actor) = TIMETAG_DIARY_SONG_EVAN_PART1;
                         break;
 
-                    case 0x122E:
+                    case 0x122E: // Mikau diary part 3
                         func_80152434(play, 0x40);
-                        this->actionFunc = func_80ACA3C0;
-                        this->actor.home.rot.z = 1;
+                        this->actionFunc = EnTimeTag_Diary_TeachEvanSongSnippets;
+                        TIMETAG_DIARY_SONG(&this->actor) = TIMETAG_DIARY_SONG_EVAN_PART2;
+                        break;
+
+                    default:
                         break;
                 }
             }
             break;
 
         case TEXT_STATE_CLOSING:
-            this->actionFunc = func_80ACA5F8;
+            this->actionFunc = EnTimeTag_Diary_Wait;
             break;
     }
 
-    if (this->actor.home.rot.x != 0) {
+    if (TIMETAG_DIARY_TIMER(&this->actor) != 0) {
         if (this->actor.cutscene == -1) {
-            this->actor.home.rot.x = 0;
+            TIMETAG_DIARY_TIMER(&this->actor) = 0;
         } else if (ActorCutscene_GetCurrentIndex() == 0x7C) {
             ActorCutscene_Stop(0x7C);
             ActorCutscene_SetIntentToPlay(this->actor.cutscene);
         } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
             ActorCutscene_Start(this->actor.cutscene, &this->actor);
-            this->actor.home.rot.x = 0;
+            TIMETAG_DIARY_TIMER(&this->actor) = 0;
         } else {
             ActorCutscene_SetIntentToPlay(this->actor.cutscene);
         }
     }
 }
 
-void func_80ACA5F8(EnTimeTag* this, PlayState* play) {
+void EnTimeTag_Diary_Wait(EnTimeTag* this, PlayState* play) {
     if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
         if (gSaveContext.save.playerForm == PLAYER_FORM_ZORA) {
-            if (ENTIMETAG_GET_SWITCHFLAG(&this->actor) == 1) {
+            if (TIMETAG_DIARY_GET_TYPE(&this->actor) == TIMETAG_DIARY_LULU) {
                 Message_StartTextbox(play, 0x101C, &this->actor);
             } else {
+                // TIMETAG_DIARY_MIKAU
                 Message_StartTextbox(play, 0x122B, &this->actor);
             }
-            this->actor.home.rot.x = 1;
+            TIMETAG_DIARY_TIMER(&this->actor) = 1;
         } else {
+            // unable to read Zora script
             Message_StartTextbox(play, 0x122A, &this->actor);
+
+            //! FAKE:
```

Comment:
Do you happen to have a decomp.me link to put here? I did that for a fake match in EnGrasshopper, for example.

Reply:
Added